velox/expression/CastExpr.cpp

```diff
@@ -64,15 +75,102 @@ void applyCastKernel(
   }
 }
 
-std::string makeErrorMessage(
+VectorPtr castFromDate(
+    const SelectivityVector& rows,
     const BaseVector& input,
-    vector_size_t row,
+    exec::EvalCtx& context,
     const TypePtr& toType) {
-  return fmt::format(
-      "Failed to cast from {} to {}: {}.",
-      input.type()->toString(),
-      toType->toString(),
-      input.toString(row));
+  VectorPtr castResult;
+  context.ensureWritable(rows, toType, castResult);
+  (*castResult).clearNulls(rows);
+
+  auto* inputFlatVector = input.as<SimpleVector<int32_t>>();
+  switch (toType->kind()) {
+    case TypeKind::VARCHAR: {
+      auto* resultFlatVector = castResult->as<FlatVector<StringView>>();
+      context.applyToSelectedNoThrow(rows, [&](int row) {
+        try {
+          auto output = DATE()->toString(inputFlatVector->valueAt(row));
+          auto writer = exec::StringWriter<>(resultFlatVector, row);
+          writer.resize(output.size());
+          if (output.size()) {
+            std::memcpy(writer.data(), output.data(), output.size());
+          }
+          writer.finalize();
+        } catch (const VeloxUserError& ue) {
+          VELOX_USER_FAIL(
+              makeErrorMessage(input, row, toType) + " " + ue.message());
```

Comment:
It's better to pass the `ue` to makeErrorMessage and use `fmt::format`.

Reply:
same as above.